k4FWCore/include/k4FWCore/MetaDataHandle.h

```diff
@@ -0,0 +1,101 @@
+#ifndef K4FWCORE_METADATAHANDLE_H
+#define K4FWCORE_METADATAHANDLE_H
+
+// GAUDI
+#include "GaudiAlg/GaudiAlgorithm.h"
+
+#include "podio/GenericParameters.h"
+#include "k4FWCore/PodioDataSvc.h"
+
+#include "GaudiKernel/MsgStream.h"
+
+#include <type_traits>
+
+template <typename T> class MetaDataHandle {
+public:
+  MetaDataHandle();
+  MetaDataHandle(const std::string& descriptor, Gaudi::DataHandle::Mode a, Algorithm* fatherAlg);
+  MetaDataHandle(const Gaudi::DataHandle& handle, const std::string& descriptor, Gaudi::DataHandle::Mode a, Algorithm* fatherAlg);
+  ~MetaDataHandle();
+
+  const T get();
+  void put(T);
+  
+private:
+  std::string fullDescriptor();
+
+private:
+  ServiceHandle<IDataProviderSvc> m_eds;
+  std::string                     m_descriptor;
+  PodioDataSvc*                   m_podio_data_service{nullptr};
+  Algorithm*                      m_fatherAlg{nullptr};
```

Comment:
Does this have to be an `Algorithm*`? Resp. is there another way of reporting errors to upstream? This seems to be the only use of the `m_fatherAlg`. In `k4MarlinWrapper` it is necessary to use this inside of a `GaudiTool`, which currently doesn't work because `this` is not an `Algorithm*`.